src/init.cpp

```diff
@@ -342,6 +343,7 @@ void DynamicDataLoader::unload_data()
     SNIPPET.clear_snippets();
     vehicle_prototype::reset();
     vpart_info::reset();
+    clear_vehicle_null_part();
```

Comment:
How would at static method of vpart_info get a reference the vehicle_part objects that need their cache cleared if I want to implement your first solution?